### PR TITLE
Fixes to make it work with GWT 2.8.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by bauna**
The important changes are in classes: GQuery and LazyGenerator

[Review on Reviewable](https://reviewable.io/reviews/arcbees/gwtquery/351)

### CODE REVIEW

File: gwtquery-core/src/main/java/com/google/gwt/query/client/GQuery.java [hunk 1]
```diff
@@ -225,7 +225,7 @@ public TagWrapper(int wrapDepth, String preWrap, String postWrap) {
    * Wrap a GQuery around the element of a Function callback.
    */
   public static GQuery $(Function f) {
-    return $(f.getElement());
+    return $((Element) f.getElement());
```
Comment (bauna) [inline gwtquery-core/src/main/java/com/google/gwt/query/client/GQuery.java]: With a java8 compiler this call is ambiguous.

File: gwtquery-core/src/main/java/com/google/gwt/query/rebind/LazyGenerator.java [hunk 2]
```diff
@@ -207,7 +207,7 @@ private void generateDoneMethod(SourceWriter sw, JClassType nonLazyType,
     }
 
     sw.println(
-        "ctx = GQuery.$(getElement()).as(" + nonLazyType.getQualifiedSourceName() + "."
+        "ctx = GQuery.$((Element) getElement()).as(" + nonLazyType.getQualifiedSourceName() + "."
```
Comment (bauna) [inline gwtquery-core/src/main/java/com/google/gwt/query/rebind/LazyGenerator.java]: With a java8 compiler this call is ambiguous.

File: samples/src/main/java/gwtquery/samples/client/GwtQueryBenchModule.java [hunk 3]
```diff
@@ -503,7 +517,7 @@ private void moveHorses(Benchmark[] b, int row, double[] totalTimes) {
     ArrayList<Benchmark> bs = new ArrayList<Benchmark>();
     for (Element e : $("input", selectPanel.getElement()).elements()) {
       String val = $(e).val().replaceAll(" .*$", "");
-      if ($(e).prop("checked")) {
+      if ($(e).<Boolean>prop("checked")) {
```
Comment (bauna) [inline samples/src/main/java/gwtquery/samples/client/GwtQueryBenchModule.java]: With a java8 compiler this call is ambiguous.

**Comment by manolo** [comment]
This issue is already fixed in #349 using a different approach avoiding to force cast.
Could you check if everything works fine in master for you?

Anyway, thanks for contributing.
- Manolo

**Comment by bauna** [comment]
I have tested GWTQuery v1.4.4-SNAPSHOT with GWTChosen v2.2.0-SNAPSHOT
They compile fine under GWT 2.8.0-SNAPSHOT but GWTChosen behave different, i.e. com.arcbees.chosen.client.gwt.ChosenListBox.forceRedraw() stop working in my app.
I'll try to make more test and let you know

**Pull request closed without merging**